seismicpro/src/seismic_batch.py

```diff
@@ -1377,3 +1377,48 @@ def shift_pick_phase(self, index, src, src_traces, dst=None, shift=1.5, threshol
         x += n_skip
         getattr(self, dst)[pos] = x
         return self
+
+    @action
+    @inbatch_parallel(init='_init_component')
+    def _load_from_index(self, index, *args, key, dst, **kwargs):
+        """Load values from index dataframe to a component
+        """
+        _ = args, kwargs
+        pos = self.get_pos(None, None, index)
+        getattr(self, dst)[pos] = self.index.get_df(index)[key].values
```

Comment:
```suggestion
        getattr(self, dst)[pos] = self.index.get_df([index])[key].values
```
make it compatible with `TraceIndex`